python/ray/tests/BUILD

```diff
@@ -1,23 +1,30 @@
 load("//bazel:python.bzl", "py_test_module_list")
 
-SRCS = [] + select({
-    "@bazel_tools//src/conditions:windows": glob([
-        # TODO(mehrdadn): This should be added for all platforms once resulting errors are fixed
-        "**/conftest.py",
-    ]),
-    "//conditions:default": [],
-})
+py_library(
+  name = "conftest",
+  srcs = glob(["**/conftest.py"]),
+  visibility = [
+    "//python/ray/tests:__subpackages__",
```

Comment:
What about the libraries that depend on these fixtures in `conftest`, e.g. `python/ray/data/tests`?

Reply:
Thanks for pointing out this. I've taken a closer look at it and //python/ray/data:tests/test_* should not be working at all (missing the deps to //python/ray/tests:conftest. 🤔 I'm investigating why it passed on CI.

Reply:
Ugh. CI [runs](https://github.com/ray-project/ray/blob/92efc85b3b4a2a62eb14ae7c1d6d5ce6ff8afa82/ci/ci.sh#L369) `pip install -e .`, which creates a symbolic link from inside site-packages to the working copy. With this it is possible to import everything without making Bazel aware of these dependencies. Other changes in this PR is needed because they didn't `import` from conftest-s, and pytest (rightfully) isn't recursing into site-packages to collect fixtures, cause they are usually considered system dependencies. OTOH if it is explicitly `import`-ed in the tests pytest happily uses them.

Anyway, I've added explicit conftest target and deps to python/ray/data/BUILD to make it more clear.

The issue seems more widespread though, for example if you run a Dataset test with `bazel test`, edit `python/ray/data/dataset.py`, and `bazel test` again, bazel decides to NOT rerunning the test because it isn't seeing the dependency between tests and sources. There is a //:ray_lib target documented as:

```
# This is a dummy test dependency that causes the python tests to be
# re-run if any of these files changes.
py_library(
    name = "ray_lib",
    srcs = glob(
        ["python/ray/**/*.py"],
        exclude = ["python/ray/tests/*.py"],
    ),
    visibility = ["__subpackages__"],
)
```

But, as you have noted, glob won't match across package boundary so it doesn't catch files under `python/ray/data`.

I guess `bazel test` (with caches) does not work for daily development and is only used in CI then.

Reply:
That makes perfect sense even if quite an unfortunate state of things, thank you for digging into that!

>Anyway, I've added explicit conftest target and deps to python/ray/data/BUILD to make it more clear.

That sounds good, thank you for making that dependency explicit even if it's not currently required.

>I guess bazel test (with caches) does not work for daily development and is only used in CI then.

Yep, our current Bazel setup is not set up as it should be, I've mentioned this internally in the past. Since most contributors use `pip install -e .` + pytest locally, when developing, there hasn't been a lot of motivation to make the local (or general) Bazel story better, although I would personally like to see it get there to align the local and CI testing stories, since this divergence makes it easy for discrepancies/bugs to creep in.